File: 933-increasing-order-search-tree/increasing-order-search-tree.py
class Solution:
    def increasingBST(self, root: TreeNode) -> TreeNode:
        dummy = curr =TreeNode(None)
        def dfs(root):
            if not root: return
            nonlocal curr
            dfs(root.left)
            curr.right = root
            curr = root
            curr.left = None
            dfs(root.right)
            
        dfs(root)
        return dummy.right

File: 933-increasing-order-search-tree/increasing-order-search-tree.cpp
class Solution {
public:
    void inorder(TreeNode* root , vector<int>& elem){
        if(root == nullptr)
        return ;

        inorder(root->left,elem);
        elem.push_back(root->val);
        inorder(root->right,elem);
    }
    TreeNode* increasingBST(TreeNode* root) {
        vector<int> elem;
        inorder(root,elem);
        
        TreeNode* temp = new TreeNode(0);
        TreeNode* curr = temp;

        for(auto x : elem){
            curr->right = new TreeNode(x);
            curr = curr->right;
        }
        return temp->right;
    }
};